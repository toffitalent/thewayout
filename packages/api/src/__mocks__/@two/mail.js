const send = jest.fn().mockResolvedValue(undefined);

module.exports = {
  send,
};
