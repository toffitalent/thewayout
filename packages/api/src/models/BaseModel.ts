import { PaginateQueryBuilder, Serialize } from '@disruptive-labs/objection-plugins';
import { Model, ModelOptions, QueryContext } from 'objection';
import { knex } from '../db';

// Bind knex instance to objection
Model.knex(knex);

export class BaseModel extends Serialize(Model) {
  static QueryBuilder = PaginateQueryBuilder;
  QueryBuilderType!: PaginateQueryBuilder<this>;

  static useLimitInFirst = true;

  createdAt!: Date;
  updatedAt!: Date;

  async $beforeInsert(queryContext: QueryContext) {
    await super.$beforeInsert(queryContext);

    this.createdAt = new Date();
    this.updatedAt = this.createdAt;
  }

  async $beforeUpdate(opt: ModelOptions, queryContext: QueryContext) {
    await super.$beforeUpdate(opt, queryContext);

    this.updatedAt = new Date();
  }
}
