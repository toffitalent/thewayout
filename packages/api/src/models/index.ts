export * from './App';
export * from './AppClient';
export * from './AppToken';
export * from './AppUser';
export * from './BaseModel';
export * from './Client';
export * from './Employer';
export * from './Job';
export * from './JobApplication';
export * from './Rsp';
export * from './RspAccount';
export * from './RspInvitation';
export * from './serializers';
export * from './User';
export {
  CheckViolationError,
  ConstraintViolationError,
  DataError,
  DBError,
  ForeignKeyViolationError,
  NotFoundError,
  NotNullViolationError,
  raw,
  ref,
  UniqueViolationError,
  ValidationError,
} from 'objection';
