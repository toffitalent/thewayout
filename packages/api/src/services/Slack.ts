import axios from 'axios';
import config from '@two/config';
import { Sentry } from './Sentry';

export async function alert(text: string) {
  const url = config.get('slack.alerts');
  if (!url) return;

  await axios
    .post(url, {
      text,
      blocks: [{ type: 'section', text: { type: 'mrkdwn', text } }],
    })
    .catch(Sentry.captureException);
}
