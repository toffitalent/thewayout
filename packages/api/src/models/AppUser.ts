import type { RelationMappingsThunk } from 'objection';
import { App } from './App';
import { BaseModel } from './BaseModel';
import { AppUserSerializer } from './serializers';
import { User } from './User';

export class AppUser extends BaseModel {
  static tableName = 'app_users';
  static Serializer = AppUserSerializer;

  readonly id!: string;
  appId!: string;
  userId!: string;
  scope!: string[];

  app?: App;
  user?: User;

  static relationMappings: RelationMappingsThunk = () => ({
    app: {
      relation: BaseModel.BelongsToOneRelation,
      modelClass: App,
      join: {
        from: 'app_users.appId',
        to: 'apps.id',
      },
    },

    user: {
      relation: BaseModel.BelongsToOneRelation,
      modelClass: User,
      join: {
        from: 'app_users.userId',
        to: 'users.id',
      },
    },
  });
}
