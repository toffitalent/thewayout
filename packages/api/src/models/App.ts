import type { RelationMappingsThunk } from 'objection';
import { AppClient } from './AppClient';
import { AppToken } from './AppToken';
import { AppUser } from './AppUser';
import { BaseModel } from './BaseModel';
import { AppSerializer } from './serializers';

export class App extends BaseModel {
  static tableName = 'apps';
  static Serializer = AppSerializer;

  readonly id!: string;
  name!: string;
  userId!: string | null;

  clients?: AppClient[];
  tokens?: AppToken[];
  users?: AppUser[];

  static relationMappings: RelationMappingsThunk = () => ({
    clients: {
      relation: BaseModel.HasManyRelation,
      modelClass: AppClient,
      join: {
        from: 'apps.id',
        to: 'app_clients.appId',
      },
    },

    tokens: {
      relation: BaseModel.HasManyRelation,
      modelClass: AppToken,
      join: {
        from: 'apps.id',
        to: 'app_tokens.appId',
      },
    },

    users: {
      relation: BaseModel.HasManyRelation,
      modelClass: AppUser,
      join: {
        from: 'apps.id',
        to: 'app_users.appId',
      },
    },
  });
}
