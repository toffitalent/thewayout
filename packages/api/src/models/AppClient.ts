import type { RelationMappingsThunk } from 'objection';
import { AuthGrantType } from '@two/shared';
import { App } from './App';
import { AppToken } from './AppToken';
import { BaseModel } from './BaseModel';

export class AppClient extends BaseModel {
  static tableName = 'app_clients';

  readonly id!: string;
  appId!: string;
  name!: string;
  secret!: string;
  grantType!: AuthGrantType;
  redirectUri!: string | null;

  app?: App;
  tokens?: AppToken[];

  static relationMappings: RelationMappingsThunk = () => ({
    app: {
      relation: BaseModel.BelongsToOneRelation,
      modelClass: App,
      join: {
        from: 'app_clients.appId',
        to: 'apps.id',
      },
    },

    tokens: {
      relation: BaseModel.HasManyRelation,
      modelClass: AppToken,
      join: {
        from: 'app_clients.id',
        to: 'app_tokens.clientId',
      },
    },
  });
}
