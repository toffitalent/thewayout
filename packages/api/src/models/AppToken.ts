import type { RelationMappingsThunk } from 'objection';
import { App } from './App';
import { AppClient } from './AppClient';
import { BaseModel } from './BaseModel';
import { User } from './User';

export class AppToken extends BaseModel {
  static tableName = 'app_tokens';

  readonly id!: string;
  appId!: string;
  clientId!: string;
  userId!: string;
  token!: Buffer;
  self!: boolean;

  app?: App;
  client?: AppClient;
  user?: User;

  static relationMappings: RelationMappingsThunk = () => ({
    app: {
      relation: BaseModel.BelongsToOneRelation,
      modelClass: App,
      join: {
        from: 'app_tokens.appId',
        to: 'apps.id',
      },
    },

    client: {
      relation: BaseModel.BelongsToOneRelation,
      modelClass: AppClient,
      join: {
        from: 'app_tokens.clientId',
        to: 'app_clients.id',
      },
    },

    user: {
      relation: BaseModel.BelongsToOneRelation,
      modelClass: User,
      join: {
        from: 'app_tokens.userId',
        to: 'users.id',
      },
    },
  });
}
