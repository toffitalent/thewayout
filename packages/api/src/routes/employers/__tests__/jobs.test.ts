import { faker } from '@faker-js/faker';
import { JobApplicationStatus } from '@two/shared';
import { Client, Employer } from '@app/models';
import * as Slack from '@app/services/Slack';
import {
  auth,
  createFakeClientProfile,
  createFakeEmployer,
  fixtures,
  request,
  resetDb,
} from '@test';

jest.mock('@app/services/Slack');

describe('Routes > Employers > jobs', () => {
  beforeEach(async () => {
    await resetDb();

    faker.seed(1);
  });

  describe('GET /api/v1/employers/:id/jobs', () => {
    test('return jobs', async () => {
      const res = await request()
        .get(`/api/v1/employers/${fixtures.user6.id}/jobs`)
        .use(auth(fixtures.user6.id, ['employer']));

      expect(res.status).toBe(200);
      expect(res.body).toMatchSnapshot();
    });

    test('wrong user id - return 403', async () => {
      const res = await request()
        .get(`/api/v1/employers/${fixtures.user4.id}/jobs`)
        .use(auth(fixtures.user5.id, ['employer']));

      expect(res.status).toBe(403);
    });

    test('returns a resource not found response for employer without profile', async () => {
      const res = await request()
        .get(`/api/v1/employers/${fixtures.user5.id}/jobs`)
        .use(auth(fixtures.user5.id, ['employer'])); // user5 is employer without profile

      expect(res.status).toBe(400);
    });
  });

  describe('GET /api/v1/employers/:id/jobs/:jobId/applications', () => {
    test('returns job applications', async () => {
      const res = await request()
        .get(`/api/v1/employers/${fixtures.user6.id}/jobs/${fixtures.job1.id}/applications`)
        .use(auth(fixtures.user6.id, ['employer']));

      expect(res.status).toBe(200);
      expect(res.body).toMatchSnapshot();
    });

    test('returns job applications with page', async () => {
      const res = await request()
        .get(`/api/v1/employers/${fixtures.user6.id}/jobs/${fixtures.job1.id}/applications`)
        .query({ page: 1 })
        .use(auth(fixtures.user6.id, ['employer']));

      expect(res.status).toBe(200);
      expect(res.body).toMatchSnapshot();
    });

    test('returns 403 when the job is not owned by the employer', async () => {
      const res = await request()
        .get(`/api/v1/employers/${fixtures.user6.id}/jobs/${fixtures.job1.id}/applications`)
        .use(auth(fixtures.user5.id, ['employer']));

      expect(res.status).toBe(403);
    });
  });

  describe('GET /api/v1/employers/:id/jobs/:jobId/applications/:clientId', () => {
    test('returns client information', async () => {
      const res = await request()
        .get(
          `/api/v1/employers/${fixtures.employer1.id}/jobs/${fixtures.job1.id}/applications/${fixtures.clientProfile1.id}`,
        )
        .use(auth(fixtures.user6.id, ['employer']));

      expect(res.status).toBe(200);
      expect(res.body).toMatchSnapshot();
    });

    test('not returns client information if not apply to employer job', async () => {
      const newClient = await Client.query()
        .insert(createFakeClientProfile({ userId: fixtures.user2.id }))
        .returning('*');
      const res = await request()
        .get(
          `/api/v1/employers/${fixtures.user6.id}/jobs/${fixtures.job1.id}/applications/${newClient.id}`,
        )
        .use(auth(fixtures.user6.id, ['employer']));

      expect(res.status).toBe(400);
    });
  });

  describe('PATCH /api/v1/employers/:id/jobs/:jobId/applications/:clientId', () => {
    test('change application status', async () => {
      const res = await request()
        .patch(
          `/api/v1/employers/${fixtures.employer1.id}/jobs/${fixtures.job1.id}/applications/${fixtures.application1.id}`,
        )
        .send({ status: JobApplicationStatus.interview })
        .use(auth(fixtures.user6.id, ['employer']));

      expect(res.status).toBe(200);
      expect(res.body).toMatchSnapshot();
    });

    test('returns 403 when the job is not owned by the employer', async () => {
      const newEmployer = await Employer.query().insert(
        createFakeEmployer({ userId: fixtures.user5.id }),
      );
      const res = await request()
        .patch(
          `/api/v1/employers/${newEmployer.id}/jobs/${fixtures.job1.id}/applications/${fixtures.application1.id}`,
        )
        .send({ status: JobApplicationStatus.interview })
        .use(auth(fixtures.user5.id, ['employer']));

      expect(res.status).toBe(403);
    });

    test('returns 403 when the employer is not authorized', async () => {
      const res = await request()
        .patch(
          `/api/v1/employers/${fixtures.employer1.id}/jobs/${fixtures.job1.id}/applications/${fixtures.application1.id}`,
        )
        .send({ status: JobApplicationStatus.interview })
        .use(auth(fixtures.user5.id, ['employer']));

      expect(res.status).toBe(403);
    });

    test('return 402 when exceeded limit of profiles uncloak', async () => {
      await request()
        .patch(
          `/api/v1/employers/${fixtures.employer1.id}/jobs/${fixtures.job1.id}/applications/${fixtures.application1.id}`,
        )
        .send({ status: JobApplicationStatus.interview })
        .use(auth(fixtures.user6.id, ['employer']));
      await request()
        .patch(
          `/api/v1/employers/${fixtures.employer1.id}/jobs/${fixtures.job2.id}/applications/${fixtures.application2.id}`,
        )
        .send({ status: JobApplicationStatus.interview })
        .use(auth(fixtures.user6.id, ['employer']));
      const res1 = await request()
        .patch(
          `/api/v1/employers/${fixtures.employer1.id}/jobs/${fixtures.job3.id}/applications/${fixtures.application3.id}`,
        )
        .send({ status: JobApplicationStatus.interview })
        .use(auth(fixtures.user6.id, ['employer']));
      expect(res1.status).toBe(200);

      const res2 = await request()
        .patch(
          `/api/v1/employers/${fixtures.employer1.id}/jobs/${fixtures.job4.id}/applications/${fixtures.application4.id}`,
        )
        .send({ status: JobApplicationStatus.interview })
        .use(auth(fixtures.user6.id, ['employer']));
      expect(res2.status).toBe(402);
      expect(Slack.alert).toBeCalledWith(
        `User ${fixtures.user6.id} exceeded limit of profile uncloak`,
      );
    });
  });
});
