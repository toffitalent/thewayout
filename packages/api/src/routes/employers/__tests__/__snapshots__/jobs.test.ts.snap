// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Routes > Employers > jobs GET /api/v1/employers/:id/jobs return jobs 1`] = `
[
  {
    "applicationsCount": 1,
    "createdAt": "<Date>",
    "department": "Customer Service",
    "description": "distinctio iure deserunt",
    "id": "01817e24-0d42-7c78-8c7b-6f6b5529a2d9",
    "location": null,
    "offensesTypes": [
      "drugs",
      "motorVehicle",
      "propertyDamage",
      "sexual",
      "theft",
      "violent",
    ],
    "pendingApplicationsCount": 1,
    "salaryOptions": {
      "bonuses": [],
      "description": "Nemo eveniet pariatur temporibus aperiam aliquam. Nostrum dignissimos dolorum ipsum facilis sit autem soluta pariatur. Repellendus adipisci officiis veniam adipisci est temporibus necessitatibus. Repellat repellendus quas temporibus libero. Explicabo exercitationem porro dignissimos rerum. Dolorum voluptatem sunt tenetur eum.
Ab beatae doloribus. Quia molestias itaque fugiat quae aspernatur inventore tempore. Reiciendis sunt aperiam reprehenderit. Cum repellat quas quas recusandae iure. Reprehenderit reiciendis officia minus pariatur unde inventore ad. Ipsam aperiam quisquam quam rerum atque sapiente accusamus vel.
Animi non ipsam quo sapiente in veritatis quidem nostrum aliquam. Ad eligendi expedita iusto corporis ad magnam magnam. Ut sit impedit. Voluptatibus ducimus perferendis id recusandae culpa.",
      "max": "40",
      "min": "30",
      "salary": "hourly",
    },
    "status": "active",
    "title": "Dynamic Optimization Supervisor",
    "typeOfWork": "hybrid",
    "updatedAt": "<Date>",
    "veteranOrJustice": [
      "justiceImpacted",
      "veteran",
    ],
    "workingTime": "fullTime",
  },
  {
    "applicationsCount": 1,
    "createdAt": "<Date>",
    "department": "Customer Service",
    "description": "Fuga necessitatibus quam assumenda aperiam eaque.",
    "id": "01817e24-0d42-7bb3-b8f7-a5f897774d1a",
    "location": null,
    "offensesTypes": [
      "drugs",
      "propertyDamage",
      "sexual",
      "theft",
      "whiteCollar",
    ],
    "pendingApplicationsCount": 1,
    "salaryOptions": {
      "bonuses": [],
      "description": "Eius et corrupti odit cumque magni modi cupiditate voluptates culpa.",
      "max": "40",
      "min": "30",
      "salary": "hourly",
    },
    "status": "active",
    "title": "Regional Infrastructure Strategist",
    "typeOfWork": "hybrid",
    "updatedAt": "<Date>",
    "veteranOrJustice": [
      "justiceImpacted",
      "veteran",
    ],
    "workingTime": "fullTime",
  },
  {
    "applicationsCount": 1,
    "createdAt": "<Date>",
    "department": "Customer Service",
    "description": "Voluptatibus est officia.",
    "id": "01817e24-0d42-7b85-8607-60e9802e5cb4",
    "location": null,
    "offensesTypes": [
      "motorVehicle",
      "propertyDamage",
      "theft",
      "violent",
      "whiteCollar",
    ],
    "pendingApplicationsCount": 1,
    "salaryOptions": {
      "bonuses": [],
      "description": "Ab sint corporis corporis fugit quidem. Praesentium illo tempore. Velit adipisci ab eius nisi nisi cum corporis.",
      "max": "40",
      "min": "30",
      "salary": "hourly",
    },
    "status": "active",
    "title": "Senior Factors Designer",
    "typeOfWork": "hybrid",
    "updatedAt": "<Date>",
    "veteranOrJustice": [
      "justiceImpacted",
      "veteran",
    ],
    "workingTime": "fullTime",
  },
  {
    "applicationsCount": 3,
    "createdAt": "<Date>",
    "department": "Customer Service",
    "description": "Delectus pariatur quae accusantium rerum consequatur. Veritatis id quisquam debitis temporibus quibusdam placeat ex saepe. Consequatur cum sint iusto sequi ducimus autem blanditiis libero. Itaque asperiores laudantium cumque. Ab molestias assumenda illum quas iste assumenda odit. Optio fuga reiciendis fuga.",
    "id": "01817e24-0d42-7b06-943b-43c05c9d4a6e",
    "location": null,
    "offensesTypes": [
      "drugs",
      "motorVehicle",
      "propertyDamage",
      "theft",
      "violent",
      "whiteCollar",
    ],
    "pendingApplicationsCount": 3,
    "salaryOptions": {
      "bonuses": [],
      "description": "quo",
      "max": "40",
      "min": "30",
      "salary": "hourly",
    },
    "status": "active",
    "title": "Central Quality Liaison",
    "typeOfWork": "hybrid",
    "updatedAt": "<Date>",
    "veteranOrJustice": [
      "justiceImpacted",
      "veteran",
    ],
    "workingTime": "fullTime",
  },
]
`;

exports[`Routes > Employers > jobs GET /api/v1/employers/:id/jobs/:jobId/applications returns job applications 1`] = `
[
  {
    "clientId": "01817e24-0d43-7c3d-aafb-617689d4b482",
    "email": "k****@*****.***",
    "firstName": "K****",
    "id": "01817e24-0d43-7c7c-b0ae-ac2a3be5cfa3",
    "jobId": "01817e24-0d42-7b06-943b-43c05c9d4a6e",
    "lastName": "B****",
    "status": "applied",
  },
  {
    "clientId": "01817e24-0d43-7c27-badf-253771388861",
    "email": "a****@*****.***",
    "firstName": "A****",
    "id": "01817e24-0d43-7c78-b13a-16bc4a5d1dcb",
    "jobId": "01817e24-0d42-7b06-943b-43c05c9d4a6e",
    "lastName": "H****",
    "status": "applied",
  },
  {
    "clientId": "01817e24-0d42-764f-8380-cabc31603866",
    "email": "a****@*****.***",
    "firstName": "G****",
    "id": "01817e24-0d42-7ce0-83e7-a85d4ca11747",
    "jobId": "01817e24-0d42-7b06-943b-43c05c9d4a6e",
    "lastName": "W****",
    "status": "applied",
  },
]
`;

exports[`Routes > Employers > jobs GET /api/v1/employers/:id/jobs/:jobId/applications returns job applications with page 1`] = `[]`;

exports[`Routes > Employers > jobs GET /api/v1/employers/:id/jobs/:jobId/applications/:clientId returns client information 1`] = `
{
  "address": "*****",
  "applicationId": "01817e24-0d42-7ce0-83e7-a85d4ca11747",
  "applicationStatus": "applied",
  "city": "*****",
  "education": [
    {
      "areaOfStudy": "Test",
      "degree": "Test",
      "description": "Voluptatem quasi natus corporis minus vel ullam non autem. Corrupti quas cupiditate doloremque beatae velit vel. Ad earum cupiditate iusto debitis illo.",
      "schoolIssuer": "Test",
      "startYear": "2010",
      "yearEarned": "2015",
    },
  ],
  "email": "a****@*****.***",
  "experience": [
    "carpentry",
    "education",
  ],
  "firstName": "G****",
  "id": "01817e24-0d42-764f-8380-cabc31603866",
  "languages": [
    {
      "language": "english",
      "level": "professional",
    },
    {
      "language": "spanish",
      "level": "basic",
    },
  ],
  "lastName": "W****",
  "license": [
    {
      "expirationAtMonth": null,
      "expirationAtYear": null,
      "issueAtMonth": "5",
      "issueAtYear": "2015",
      "issuingOrganization": "Test",
      "licenseName": "Test",
    },
  ],
  "personalStrengths": [
    "creative",
    "dealsWellWithChange",
    "goodWithTechnology",
    "peoplePerson",
    "managesTimeWell",
  ],
  "personalSummary": "sit",
  "postalCode": "*****",
  "questions": null,
  "relativeExperience": [
    {
      "company": "*****",
      "description": "A quos animi veritatis voluptatum. At quibusdam ipsam temporibus.",
      "endAtMonth": "3",
      "endAtYear": "2018",
      "location": "Location",
      "startAtMonth": "3",
      "startAtYear": "2015",
      "title": "Test",
    },
  ],
  "state": "*****",
  "updatedAt": "<Date>",
  "veteranOrJustice": [
    "justiceImpacted",
  ],
}
`;

exports[`Routes > Employers > jobs PATCH /api/v1/employers/:id/jobs/:jobId/applications/:clientId change application status 1`] = `
{
  "client": {
    "address": "Lynch Island",
    "age": "25-34",
    "city": "South Madgeside",
    "disability": false,
    "education": [
      {
        "areaOfStudy": "Test",
        "degree": "Test",
        "description": "Voluptatem quasi natus corporis minus vel ullam non autem. Corrupti quas cupiditate doloremque beatae velit vel. Ad earum cupiditate iusto debitis illo.",
        "schoolIssuer": "Test",
        "startYear": "2010",
        "yearEarned": "2015",
      },
    ],
    "ethnicity": "eastAsian",
    "experience": [
      "carpentry",
      "education",
    ],
    "firstName": "Gilbert",
    "gender": "female",
    "id": "01817e24-0d42-764f-8380-cabc31603866",
    "justiceStatus": "freeWorld",
    "languages": [
      {
        "language": "english",
        "level": "professional",
      },
      {
        "language": "spanish",
        "level": "basic",
      },
    ],
    "lastName": "Witting",
    "license": [
      {
        "expirationAtMonth": null,
        "expirationAtYear": null,
        "issueAtMonth": "5",
        "issueAtYear": "2015",
        "issuingOrganization": "Test",
        "licenseName": "Test",
      },
    ],
    "maritalStatus": "married",
    "orientation": "heterosexual",
    "personalStrengths": [
      "creative",
      "dealsWellWithChange",
      "goodWithTechnology",
      "peoplePerson",
      "managesTimeWell",
    ],
    "personalSummary": "sit",
    "phone": null,
    "postalCode": "98638",
    "referredBy": "newsMedia",
    "relativeExperience": [
      {
        "company": "Test",
        "description": "A quos animi veritatis voluptatum. At quibusdam ipsam temporibus.",
        "endAtMonth": "3",
        "endAtYear": "2018",
        "location": "Location",
        "startAtMonth": "3",
        "startAtYear": "2015",
        "title": "Test",
      },
    ],
    "religion": "noReligion",
    "state": "AK",
    "support": [
      "basicTechSkills",
      "basicHygieneSupport",
    ],
    "updatedAt": "<Date>",
    "userId": "01817e24-0d42-7570-b157-986b4086c921",
    "veteranOrJustice": [
      "justiceImpacted",
    ],
    "veteranStatus": "notVeteran",
  },
  "clientId": "01817e24-0d42-764f-8380-cabc31603866",
  "createdAt": "<Date>",
  "id": "01817e24-0d42-7ce0-83e7-a85d4ca11747",
  "jobId": "01817e24-0d42-7b06-943b-43c05c9d4a6e",
  "status": "interview",
  "updatedAt": "<Date>",
}
`;
